src/contracts/storage/pool.rs

```diff
@@ -174,11 +174,19 @@ impl Pool {
         ))
     }
 
+    pub fn update_tick_index(&mut self, result: SwapResult, fee_tier: FeeTier) {
```

Comment:
What is the reason for defining this function? If this function is genuinely necessary, the naming should definitely be different. It updates the tick, but only in one of the possible cases